lib/install/management/appliance.go

```diff
@@ -503,7 +569,20 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 	} else {
 		// if vapp is not created, fall back to create VM under default resource pool
 		info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-			return d.session.VMFolder.CreateVM(ctx, *spec, d.vchPool, d.session.Host)
+
+			vchParentFolder, err := d.createVCHInventoryFolders(spec)
```

Comment:
Should not be in this WaitForResult. Move it out and above.

Reply:
Arguably this function should not be in this function at all. it should be in the `CreateVCH` path

Reply:
Kept it in the same function, but moved it out of the WaitForResult.